modules/develop/pages/data-transforms.adoc

```diff
@@ -0,0 +1,251 @@
+
+== Limitations
+
+- Transforms have no external access to disk or network resources. 
+- Only single record transforms are supported, not aggregations or joins. In other words, a Wasm function acts on a single record in an input topic.
+- Only a single output topic is supported.
+- Transforms have at-least-once delivery.
+- Because data transforms are powered by Wasm, transform functions can be authored in any language. However, currently libraries to author transforms are only available in xref:reference:data-transform-api.adoc[Golang^]. 
+
+== How data transforms work
+
+Within Repdanda, there's a single Raft controller that manages cluster information, including data transforms. On every shard (core), Redpanda knows what data transforms exist in the cluster, as well as metadata about the transform, such as input and output topics and environment variables. 
+
+Wasm is executed within a runtime in the broker. Data transforms take an input topic and map it to an output topic, running a virtual machine (VM) on the same shard as the input partition leader. Redpanda uses just-in-time (JIT) compilation, which compiles the function in memory, writes it to executable space, then runs the directly translated machine code. When it becomes leader of a given partition on the input topic, Redpanda spins up a Wasm VM using the JIT-compiled WebAssembly module. Redpanda writes the output, which may go to the same shard on the same machine, or to a different shard on the same machine, or to a different machine. 
+
+image::shared:wasm_architecture.png[Wasm architecture in Redpanda]
+
+The transform function processes every record produced to an input topic and returns zero or more records that are then produced to an output topic. Data transforms are applied to all partitions on an input topic. A record is processed after it has been successfully written to disk on the input topic. Because the transform happens in the background after the write finishes, the transform doesn't affect the original produced record and it doesn't block writes to the input topic. The following diagram shows that Broker A has its own disk that it persistently writes to, and Broker B also has its own disk that it persistently writes to. 
+
+image::shared:wasm_flow.png[Flow of record transform]
+
+When you deploy a data transform, Redpanda pushes the generated `.wasm` files and metadata to the cluster, which then stores and replicates the bytecode and metadata using internal broker topics. Each shard of Redpanda runs its own copy of the transform function. There is at most one runtime for a transform for any shard, and memory for each function is reserved within the broker. This is achieved with the following cluster configuration properties: 
```

Comment:
This switches to talk about how deploys work and what configuration is. Should there be a header or something to make this change of gears more clear?